src/compile/data/bin.ts

```diff
@@ -83,7 +88,7 @@ export const bin: DataComponentCompiler<Dict<VgTransform[]>> = {
     // If child doesn't have its own data source, then merge
     if (!childDataComponent.source) {
       // FIXME: current merging logic can produce redundant transforms when a field is binned for color and for non-color
-      extend(binComponent, childDataComponent.bin);
+      binComponent = {...childDataComponent.bin};
```

Comment:
The equivalent code to the old code would be

```
binComponent = {
  ...binComponent,
  ...childDataComponent.bin
}
```

That said, I think when we want to cause side effect the old `extend` method seems better.  
(I take back my comment from slack this morning.)